src/Utils.hpp

```diff
@@ -68,6 +72,23 @@ inline std::shared_ptr<BaseIO> createIO(const std::string& type,
   }
 }
 
+inline void convertFloatToInt16LE(const float* source,
+                                  void* dest,
+                                  int numSamples)
+{
+  auto maxVal = static_cast<double>(0x7fff);
+  auto intData = static_cast<char*>(dest);
```

Comment:
Can you clarify why dest is void* instead of char*? Since we use a static_cast to char* it seems that we assume that dest is always char*?

Reply:
I'm a bit confused. Why do we need to cast to char (i.e., an 8bit type) if intData is supposed be and int16?

Reply:
Hmm good question, honestly I was just adapting this original [JUCE function](https://github.com/juce-framework/JUCE/blob/46c2a95905abffe41a7aa002c70fb30bd3b626ef/modules/juce_audio_basics/buffers/juce_AudioDataConverters.cpp#L41-L64) using the default values.

I believe the intermediate cast to char was for byte-wise pointer manipulation. Since we are just using the default value for destBytesPerSample, I'm not sure if we need this anymore. 

Similarly for the casting to uint16, we might be able to skip that step since we recast to int16 in the `transformToInt16` function anyway.

Reply:
What I would suggest then is to: 1) add a docstring to describe what the function is doing and 2) add some comments or TODO items in the function itself about these details so that if/when we need to revisit the function we can understand why these different types and casts are happening they way they are. As long as the function is working as intended, I don't think it is a high priority to change this right now.

Reply:
sounds good! Added